doc/mgr/rgw.rst

```diff
@@ -18,36 +19,111 @@ RGW Realm Operations
 
 Bootstrapping RGW realm creates a new RGW realm entity, a new zonegroup,
 and a new zone. It configures a new system user that can be used for
-multisite sync operations, and returns a corresponding token. It sets
-up new RGW instances via the orchestrator.
+multisite sync operations. Under the hood this module instructs the
+orchestrator to create and deploy the corresponding RGW daemons. The module
+supports both passing the arguments in the cmd line as in the form of a spec
+file:
 
-It is also possible to create a new zone that connects to the master
-zone and synchronizes data to/from it.
+.. prompt:: bash #
+
+  rgw realm bootstrap [--realm-name] [--zonegroup-name] [--zone-name] [--port] [--placement] [--start-radosgw]
+
+The command supports providing the configuration through a spec file (`-i option`):
+
+.. prompt:: bash #
+
+  ceph rgw realm bootstrap -i myrgw.yaml
+
+Following is an example of RGW mutlisite spec file:
+
+.. code-block:: yaml
+
+  rgw_realm: myrealm
+  rgw_zonegroup: myzonegroup
+  rgw_zone: myzone
+  placement:
+    hosts:
+     - ceph-node-1
+     - ceph-node-2
+  spec:
+    rgw_frontend_port: 5500
+
+.. note:: The spec file used by RGW has the same format as the one used by cephadm. Thus,
+          the user can provide any cephadm rgw supported parameter as any other advanced
+          configuration items such as SSL certificates etc.
 
 
 Realm Credentials Token
 -----------------------
-A new token is created when bootstrapping a new realm, and also
-when creating one explicitly.  The token encapsulates
-the master zone endpoint, and a set of credentials that are associated
-with a system user.
-Removal of this token would remove the credentials, and if the corresponding
-system user has no more access keys, it is removed.
 
+User can list the available tokens for the created (or already existing) realms.
+The token is a base64 string that encapsulates the realm information and its
+master zone endpoint authentication data. Following is an example of
+the `ceph rgw realm tokens` output:
+
+.. prompt:: bash #
+
+  ceph rgw realm tokens | jq
+
+.. code-block:: json
+
+  [
+    {
+      "realm": "myrealm1",
+      "token": "ewogICAgInJlYWxtX25hbWUiOiAibXlyZWFs....NHlBTFhoIgp9"
+    },
+    {
+      "realm": "myrealm2",
+      "token": "ewogICAgInJlYWxtX25hbWUiOiAibXlyZWFs....RUU12ZDB0Igp9"
+    }
+  ]
+
+User can use the token to create and synchronize a secondary zones
+on another cluster with the master zone by using `ceph rgw zone create`
+command and proving the corresponding token.
+
+Following is an example of zone spec file:
+
+.. code-block:: yaml
+
+  rgw_realm: myrealm
+  rgw_zonegroup: myzonegroup
+  rgw_zone: my-secondary-zone
+  rgw_realm_token: <token>
+  placement:
+    hosts:
+     - ceph-node-1
+     - ceph-node-2
+  spec:
+    rgw_frontend_port: 5500
```

Comment:
@adk3798 in fact for the secondary zone creation we only need `rgw_zone` and `rgw_realm_token`. I lef the other fields because the RGW spec was checking for them. I have updated the code to only check for the token/zone in case of zone creation.